airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -201,35 +207,78 @@ def delete_collections(
         :return: if `if_error=continue` return list of collections which we failed to delete.
             if `if_error=stop` returns None.
         """
-        client = self.get_conn()
-        collection_names = (
-            [collection_names] if collection_names and isinstance(collection_names, str) else collection_names
-        )
-
-        failed_collection_list = []
-        for collection_name in collection_names:
-            try:
-                for attempt in Retrying(
-                    stop=stop_after_attempt(3),
-                    retry=(
-                        retry_if_exception(lambda exc: check_http_error_is_retryable(exc))
-                        | retry_if_exception_type(REQUESTS_EXCEPTIONS_TYPES)
-                    ),
-                ):
-                    with attempt:
-                        self.log.info(attempt)
-                        client.collections.delete(collection_name)
-            except Exception as e:
-                if if_error == "continue":
-                    self.log.error(e)
-                    failed_collection_list.append(collection_name)
-                elif if_error == "stop":
-                    raise e
+        if by_property:
+            self.delete_by_property(collection_names, by_property, contains_any, contains_all)
+        else:
+            client = self.get_conn()
+            collection_names = (
+                [collection_names]
+                if collection_names and isinstance(collection_names, str)
+                else collection_names
+            )
+            failed_collection_list = []
+            for collection_name in collection_names:
+                try:
+                    for attempt in Retrying(
+                        stop=stop_after_attempt(3),
+                        retry=(
+                            retry_if_exception(lambda exc: check_http_error_is_retryable(exc))
+                            | retry_if_exception_type(REQUESTS_EXCEPTIONS_TYPES)
+                        ),
+                    ):
+                        with attempt:
+                            self.log.info(attempt)
+                            client.collections.delete(collection_name)
+                except Exception as e:
```

Comment:
Ah sorry I miss this 🤦‍♂️ I think this needs to be changed as well
```suggestion
                except requests.exceptions.ConnectionError as e:
```